apiserver/routing/routes.go

```diff
@@ -0,0 +1,247 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	"github.com/crunchydata/postgres-operator/apiserver/backrestservice"
+	"github.com/crunchydata/postgres-operator/apiserver/backupservice"
+	"github.com/crunchydata/postgres-operator/apiserver/benchmarkservice"
+	"github.com/crunchydata/postgres-operator/apiserver/catservice"
+	"github.com/crunchydata/postgres-operator/apiserver/clusterservice"
+	"github.com/crunchydata/postgres-operator/apiserver/configservice"
+	"github.com/crunchydata/postgres-operator/apiserver/dfservice"
+	"github.com/crunchydata/postgres-operator/apiserver/failoverservice"
+	"github.com/crunchydata/postgres-operator/apiserver/labelservice"
+	"github.com/crunchydata/postgres-operator/apiserver/loadservice"
+	"github.com/crunchydata/postgres-operator/apiserver/lsservice"
+	"github.com/crunchydata/postgres-operator/apiserver/namespaceservice"
+	"github.com/crunchydata/postgres-operator/apiserver/pgbouncerservice"
+	"github.com/crunchydata/postgres-operator/apiserver/pgdumpservice"
+	"github.com/crunchydata/postgres-operator/apiserver/pgoroleservice"
+	"github.com/crunchydata/postgres-operator/apiserver/pgouserservice"
+	"github.com/crunchydata/postgres-operator/apiserver/pgpoolservice"
+	"github.com/crunchydata/postgres-operator/apiserver/policyservice"
+	"github.com/crunchydata/postgres-operator/apiserver/pvcservice"
+	"github.com/crunchydata/postgres-operator/apiserver/reloadservice"
+	"github.com/crunchydata/postgres-operator/apiserver/scheduleservice"
+	"github.com/crunchydata/postgres-operator/apiserver/statusservice"
+	"github.com/crunchydata/postgres-operator/apiserver/upgradeservice"
+	"github.com/crunchydata/postgres-operator/apiserver/userservice"
+	"github.com/crunchydata/postgres-operator/apiserver/versionservice"
+	"github.com/crunchydata/postgres-operator/apiserver/workflowservice"
+
+	"github.com/gorilla/mux"
+)
+
+// RegisterAllRoutes adds all routes supported by the apiserver to the
+// provided router
+func RegisterAllRoutes(r *mux.Router) {
+	RegisterBackrestSvcRoutes(r)
+	RegisterBackupSvcRoutes(r)
+	RegisterBenchmarkSvcRoutes(r)
+	RegisterCatSvcRoutes(r)
+	RegisterClusterSvcRoutes(r)
+	RegisterConfigSvcRoutes(r)
+	RegisterDfSvcRoutes(r)
+	RegisterFailoverSvcRoutes(r)
+	RegisterLabelSvcRoutes(r)
+	RegisterLoadSvcRoutes(r)
+	RegisterLsSvcRoutes(r)
+	RegisterNamespaceSvcRoutes(r)
+	RegisterPGBouncerSvcRoutes(r)
+	RegisterPGDumpSvcRoutes(r)
+	RegisterPGORoleSvcRoutes(r)
+	RegisterPGOUserSvcRoutes(r)
+	RegisterPGPoolSvcRoutes(r)
+	RegisterPolicySvcRoutes(r)
+	RegisterPVCSvcRoutes(r)
+	RegisterReloadSvcRoutes(r)
+	RegisterScheduleSvcRoutes(r)
+	RegisterStatusSvcRoutes(r)
+	RegisterUpgradeSvcRoutes(r)
+	RegisterUserSvcRoutes(r)
+	RegisterVersionSvcRoutes(r)
+	RegisterWorkflowSvcRoutes(r)
```

Comment:
Is there a need to have all these functions exported?

Reply:
The scenario that spawned the change only registered one service and the pattern is to allow only importing the routes of a specific "service", allowing routes to be composed on a per-service basis as might be used for testing or service-specific middleware. As I mentioned in the comments (PR? commit?), ultimately it would make sense for each service to export a registration for its handlers for better cohesion.

Reply:
Gotcha, was just a curiosity. But, yeah, that makes sense as to the motivation for the refactor. And I suppose it answers the question of 'exporting' given the goal of testing, etc. Certainly a good change for now. Though, for "later", I think moving towards the concept of 'service' and 'routing/transport' being separate/decoupled would be beneficial to those ends as well. Just a thought.